Ihamer/7468 inject noc delays (#8889)
tenstorrent/tt-metal: docs/source/tt-metalium/tools/watcher.rst

         0x00000020,0x0000001f,0x0000001e,0x0000001d,0x0000001c,0x0000001b,0x0000001a,0x00000019,
         0x00000018,0x00000017,0x00000016,0x00000015,0x00000014,0x00000013,0x00000012,0x00000011,
         0x00000010,0x0000000f,0x0000000e,0x0000000d,0x0000000c,0x0000000b,0x0000000a]

Debug Delays
------------

> Watcher can insert NOC transactions delays for debugging purposes. The delayes can be specified by

Reviewer: ```suggestion
Watcher can insert NOC transaction delays for debugging purposes. These delays can be specified by
```

transaction type and location. Environment variable `TT_METAL_WATCHER_DELAY` specifies the number
of spin loop iterations to wait for. Similarly to DPRINT, the delay can be set for all cores, or a
or a subset by setting environment variable `TT_METAL_*_DEBUG_DELAY_CORES`: x,y OR (x1,y1),(x2,y2),(x3,y3) OR (x1,y1)-(x2,y2) OR all.
The * can be one of: READ, WRITE or ATOMIC indicating whether the delays will be inserted before read, write or atomic NOC
transactions. Finally, the delay can be set for a specific RISCs (BRISC, NCRISC, TRISC0, TRISC1, TRISC2) through the
environment variable `TT_METAL_*_DEBUG_DELAY_RISCVS`: (one of: BR,NC,TR0,TR1,TR2); if not set, the delay
is applied to all RISCs.
Note that `TT_METAL_WATCHER` must be set and `TT_METAL_WATCHER_DISABLE_NOC_SANITIZE` must not be
set for the delays to be applied.

For example, the following environment variables will add a delay of 10 iterations for both READ
and WRITE transactions on BRISC core at location 0,0.

.. code-block::

>     TT_METAL_WATCHER=1 TT_METAL_WATCHER_DEBUG_DELAY=10 TT_METAL_READ_DEBUG_DELAY_CORES=0,0 TT_METAL_WRITE_DEBUG_DELAY_CORES=0,0 TT_METAL_READ_DEBUG_DELAY_RISCVS=BR TT_METAL_WRITE_DEBUG_DELAY_RISCVS=BR

Reviewer 2: Did you check on a local docs build that these environment display on new lines? I'm unsure if that's your intention.

I would also suggest writing an example command with a unit tests executable to show how to use it. If it's one thing I've learned, it's never assume your readers know what you're talking about, including as something elementary as Unix environment variables.
